esphome/components/pzemdc/sensor.py

```diff
@@ -59,6 +64,20 @@
 )
 
 
+@automation.register_action(
+    "pzemdc.reset_energy",
+    ResetEnergyAction,
+    maybe_simple_id(
+        {
+            cv.Required(CONF_ID): cv.use_id(PZEMDC),
```

Comment:
This allows auto linking if you only have 1 in yaml
```suggestion
            cv.GenerateID(CONF_ID): cv.use_id(PZEMDC),
```